### PR TITLE
[CSV-300] Return new ArrayList in CSVRecord.toList() to prevent access to private values array

### PR DESCRIPTION
**Pull request by sman-81**
This PR offers a fix to bug CSV-300.

### CODE REVIEW

File: src/main/java/org/apache/commons/csv/CSVRecord.java [hunk 1]
```diff
@@ -298,13 +299,14 @@ public Stream<String> stream() {
...
+     * Converts the values to a new {@code List}.<br>
+     * Modifications to the list do not alter this record.
      *
-     * @return a new List
+     * @return a new {@code List}
      * @since 1.9.0
      */
     public List<String> toList() {
-        return Arrays.asList(values);
+        return new ArrayList<>(Arrays.asList(values));
```
Comment (kinow) [inline src/main/java/org/apache/commons/csv/CSVRecord.java:309]: See @aherbert 's comment in the JIRA linked about returning an immutable list. Unless there's a reason for allowing users to modify the returned list, I think an immutable list would be best :+1: 

Thanks for the pull request!
-Bruno

**Comment by garydgregory** [comment]
-1 needs a unit test

File: src/main/java/org/apache/commons/csv/CSVRecord.java (hunk 1 above)
Comment (kinow) [reply src/main/java/org/apache/commons/csv/CSVRecord.java:309]: (Also needs to update the javadocs as mentioned in the JIRA comment too, and needs unit tests as pointed below by @garydgregory :+1: )

Comment (garydgregory) [reply src/main/java/org/apache/commons/csv/CSVRecord.java:309]: I don't see the point in returning a half-functional list. The list should not mangle the underlying record that's all.

**Comment by sman-81** [comment]
Thanks for your feedback. PR updated.

**Comment by garydgregory** [comment]
https://issues.apache.org/jira/browse/CSV-300

**Comment by garydgregory** [comment]
Hm, I don't see the point of a read only list. IMO if I convert the record to a list, I should be able to do whatever I want with the list. If the list is RO, then I have to make a copy to edit it which is counter intuitive and wasteful. This is a toList() convert method and not a getList() access method after all. The point of a list here IMO is that I can edit its size, unlike an array.

**Comment by sman-81** [comment]
Reason for closing pr: @garydgregory implemented his own solution in master, making my pr obsolete.

**Pull request closed without merging**